### DISCUSSION TITLE
Question about Tokenizer

### DISCUSSION THREAD

**Discussion by yiruihe** [MaartenGr/BERTopic#1610] [Q&A] [answered]
Hi! I have a question regarding the usage of the tokenizer.

While using BERTopic working with some models like [CodeT5](https://huggingface.co/Salesforce/codet5-large) or [CodeBERT](https://huggingface.co/neulab/codebert-java) from huggingface, I noticed that they need a special tokenizer. I was wondering whether BERTopic is able to handle those, or if we have to do the embedding ourselves and pass those embeddings to BERTpoic to do the following steps.

**Comment by MaartenGr** [comment]
If you are using a transformers pipeline, then the pipeline will use the tokenizer that relates to the model you are using. For example:

```python
from transformers import pipeline

# Create your representation model
generator = pipeline('text2text-generation', model='google/flan-t5-base')
```

In the example above, if you only specify the model it will also search that directory for its tokenizer. The same applies to the models you refer to.

**Reply by yiruihe** [reply]
Hi @MaartenGr, thanks for the answer! 

First I want to clarify something. The tokenizer I'm asking is at the embedding step, not the step that generates topic representation.

From what I gather, the concept of a `pipeline` is closely tied to a specific task, like `'text2text-generation'` that you mentioned. Based on different tasks, the pipeline will return different kinds of `pipeline`.  When it comes to BERTopic, the primary requirement is a model that can effectively handle embeddings. Does this suggest that the most appropriate task for models in the BERT family or CodeT5 would be ['feature-extraction'](https://huggingface.co/tasks/feature-extraction)?

**Reply by MaartenGr** [reply, marked as answer]
The `pipeline` is merely a wrapper around some basic functionality but helps to boil it down to a single line of code. 

> When it comes to BERTopic, the primary requirement is a model that can effectively handle embeddings. Does this suggest that the most appropriate task for models in the BERT family or CodeT5 would be ['feature-extraction'](https://huggingface.co/tasks/feature-extraction)?

Yes, for embeddings that would indeed be appropriate as indicated in the documentation [here](https://maartengr.github.io/BERTopic/getting_started/embeddings/embeddings.html#hugging-face-transformers). However, if you are using T5 models for text-to-text, then [this documentation page](https://maartengr.github.io/BERTopic/getting_started/representation/llm.html#transformers) might be helpful.

**Reply by yiruihe** [reply]
Ah I see! Thank you very much for your answer! I think the `base transformer` [here](https://huggingface.co/docs/transformers/v4.35.0/en/main_classes/pipelines#transformers.FeatureExtractionPipeline) confused me to some extend. 😂
